Ginger/GingerCore/Drivers/JavaDriverLib/JavaDriver.cs

```diff
@@ -2965,56 +2971,184 @@ private void GetPOMAction(PayLoad payLoad)
-            configArgs.LocateValue = payLoad.GetValueString();
 
-            if (!payLoad.Name.Equals("SwitchWindow"))
+            ElementInfo elementInfo;
+
+            if (payLoad.Name.StartsWith("html", StringComparison.OrdinalIgnoreCase))
             {
+                configArgs.LocateBy = payLoad.GetValueString();
+                configArgs.LocateValue = payLoad.GetValueString();
                 configArgs.ElementValue = payLoad.GetValueString();
+                //read payload to avoid exception
+                var optype = payLoad.GetValueString();
+                var elType = payLoad.GetValueString();
+
+                var xCord = payLoad.GetValueString();
+                var yCord = payLoad.GetValueString();
+
+
+                elementInfo = new HTMLElementInfo();
+                elementInfo.ElementType = nameof(HTMLElementInfo);
+                if (configArgs.LocateBy.Equals(nameof(eLocateBy.ByXPath)))
+                {
+                    elementInfo.XPath = configArgs.LocateValue;
+                }
+                SetHtmlElementSpecificConfiguration(payLoad, configArgs);
+                
+                var elInfo = LearnHtmlElementByXYCord(xCord, yCord);
+
+                if (elInfo != null)
+                {
+                    elInfo.ElementName = elInfo.ElementTitle + elInfo.Value;
+
+                    elementInfo = ((IWindowExplorer)this).LearnElementInfoDetails(elInfo);
+                    elementInfo.Locators.ToList().ForEach(x => x.Active = true);
+                    elInfo.Properties.Add(new ControlProperty() { Name = ElementProperty.ParentBrowserPath, Value = parentBrowserPath });
+                    configArgs.LearnedElementInfo = elInfo;
+                    return configArgs;
+                }
+
+
             }
+            else
+            {
+                configArgs.LocateBy = eLocateBy.ByXPath;
+                configArgs.LocateValue = payLoad.GetValueString();
+
+                if (!payLoad.Name.Equals("SwitchWindow"))
+                {
+                    configArgs.ElementValue = payLoad.GetValueString();
+                }
 
-            SetElementSpecificConfiguration(payLoad, configArgs);
+                elementInfo = new JavaElementInfo();
+                elementInfo.XPath = configArgs.LocateValue;
+
+                elementInfo.ElementType = nameof(JavaElementInfo);
+                SetElementSpecificConfiguration(payLoad, configArgs);
+            }
 
             configArgs.Description = string.Concat(configArgs.Operation, " ", configArgs.LocateValue);
-            configArgs.LearnedElementInfo = GetElementInfoFromActionConfiguration(configArgs);
+
+
+            configArgs.LearnedElementInfo = GetElementInfoFromActionConfiguration(configArgs, elementInfo);
 
 
             return configArgs;
         }
 
+        private void SetHtmlElementSpecificConfiguration(PayLoad payLoad, ElementActionCongifuration configArgs)
+        {
+            switch (payLoad.Name)
```

Comment:
All this code should not be in Driver. Instead it should be in Platform.
See GetPlatformActionByElementInfo and GetDefaultElementOperation in JavaPlatform class